setup.py

```diff
@@ -1,107 +1,81 @@
 #!/usr/bin/env python3
-
-from setuptools import setup
-from setuptools.command.install import install as orig_install
-
-from distutils.core import Command
+import os.path
+import stat
+import sys
 from distutils.command.build import build as orig_build
-
-from os import path
-
-from sys import platform
-
-from urllib.request import urlopen
-from http import HTTPStatus
-
+from distutils.core import Command
 from hashlib import sha512
+from http import HTTPStatus
+from urllib.request import urlopen
 
-from os import makedirs
-from os import fstat
-from os import fchmod
+from setuptools import setup
+from setuptools.command.install import install as orig_install
 
-from functools import lru_cache
+SHELLCHECK_VERSION = '0.7.0'
+PY_VERSION = '1'
 
 
 def get_arch() -> str:
-    if platform == "linux" or platform == "linux2":
+    if sys.platform.startswith('linux'):
         # TODO(rhee): detect "linux-aarch64" and "linux-armv6hf"
         return 'linux-x86_64'
-    elif platform == "darwin":
+    elif sys.platform == 'darwin':
         return 'darwin-x86_64'
-    elif platform == "win32" or platform == "win64":
+    elif sys.platform == 'win32':
         return 'exe'
     else:
-        raise RuntimeError('Unsupported platform')
+        raise ValueError('Unsupported platform')
 
 
 def get_executable_name() -> str:
-    return f'shellcheck-v{get_version()}.{get_arch()}'
-
-
-@lru_cache(maxsize=1)
-def get_version() -> str:
-    with open('VERSION', 'r') as fp:
-        return fp.read().strip()
+    return f'shellcheck-v{SHELLCHECK_VERSION}.{get_arch()}'
 
 
 def get_download_url() -> str:
-    return path.join(
-        'https://storage.googleapis.com/',
-        'shellcheck',
-        get_executable_name(),
-    )
+    return f'https://storage.googleapis.com/shellcheck/{get_executable_name()}'
```

Comment:
`os.path.join` isn't appropriate for url joining (on windows it'd put backslashes in there (!))

Reply:
oh derp, true